src/Raven.Client/Http/RavenCommand.cs

```diff
@@ -66,6 +66,12 @@ public abstract class RavenCommand<TResult>
 
         internal long FailoverTopologyEtag = -2;
 
+        internal string Etag; // for NotModified use
+
+        internal Action<HttpRequestMessage> ModifyRequest;
+
+        internal Func<string, string> ModifyUrl;
+
```

Comment:
Why is this needd?

Reply:
we need a way to pass the some headers to all shards.. 
e.g. LastKnownRaftLast header